controllers/provisioners/eks/helpers.go

```diff
@@ -97,18 +97,43 @@ func (ctx *EksInstanceGroupContext) ResolveSecurityGroups() []string {
 }
 
 func (ctx *EksInstanceGroupContext) GetBasicUserData(clusterName, args string, kubeletExtraArgs string, payload UserDataPayload, mounts []MountOpts) string {
-      osFamily := ctx.GetOsFamily()
+	var (
+		instanceGroup = ctx.GetInstanceGroup()
+		configuration = instanceGroup.GetEKSConfiguration()
+		cluster       = ctx.GetDiscoveredState().Cluster
+		apiEndpoint   = cluster.Endpoint
+		clusterCa     = cluster.CertificateAuthority.Data
+		osFamily      = ctx.GetOsFamily()
+		nodeLabels    = ctx.GetComputedLabels()
+		nodeTaints    = configuration.GetTaints()
+	)
+
 	var UserDataTemplate string
-	if strings.EqualFold(osFamily, OsFamilyWindows) {
+	switch strings.ToLower(osFamily) {
+	case OsFamilyWindows:
 		UserDataTemplate = `
 <powershell>
   [string]$EKSBinDir = "$env:ProgramFiles\Amazon\EKS"
   [string]$EKSBootstrapScriptName = 'Start-EKSBootstrap.ps1'
   [string]$EKSBootstrapScriptFile = "$EKSBinDir\$EKSBootstrapScriptName"
   & $EKSBootstrapScriptFile -EKSClusterName {{ .ClusterName }} -KubeletExtraArgs '{{ .KubeletExtraArgs }}' 3>&1 4>&1 5>&1 6>&1
-</powershell>
-		`
-	} else {
+</powershell>`
+	case OsFamilyBottleRocket:
```

Comment:
Looks like we added apiServer & certificate into our context, where in the case of linux the bootstrap.sh script polls for this.
Should we align all OS families to same approach?
Whether we query in userdata or if we use the state we collected does not matter as long as it does not cost in additional API calls - so as long as we already have the data (via DescribeCluster), should we pass the api / certificate into bootstrap.sh script? or the opposite - should we make the bottlerocket/other userdata query those values?
https://github.com/awslabs/amazon-eks-ami/blob/master/files/bootstrap.sh#L223-L241

Reply:
The reason we need this is that BottleRocket doesn't support querying these values - they're required input parameters for UserData. I think it makes sense to continue allowing the bootstrap script in the AL2 and Windows AMIs to look it up.

Reply:
Though, one argument against that is that the AMIs aren't always consistent with each other - see (https://github.com/bottlerocket-os/bottlerocket/issues/1197). Injecting information from instance-manager could help protect against AMI drift/inconsistency.

Reply:
We can probably keep this as is, might want to pass the same flags to bootstrap.sh (since it accepts these in flags as override) in the future - it will make bootstrapping faster and more reliable.